bin/tag.ml

```diff
@@ -10,22 +10,54 @@ open Dune_release
 let vcs_tag tag ~dry_run ~commit_ish ~force ~sign ~delete ~msg ~yes =
   let msg = match msg with None -> strf "Distribution %s" tag | Some m -> m in
   Vcs.get () >>= fun repo ->
+  let tag_exists = Vcs.tag_exists ~dry_run:false repo tag in
   match delete with
   | true ->
-      Prompt.confirm_or_abort ~yes ~question:(fun l ->
-          l "Delete tag %a?" Text.Pp.version tag)
-      >>= fun () ->
-      Vcs.delete_tag ~dry_run repo tag >>| fun () ->
-      App_log.success (fun m -> m "Deleted tag %a" Text.Pp.version tag)
+      if tag_exists then
+        Vcs.tag_points_to_commit repo tag commit_ish
+        >>= fun tag_points_to_commit ->
+        let question =
+          if tag_points_to_commit then
+            Prompt.confirm_or_abort ~yes ~question:(fun l ->
+                l "Delete tag %a?" Text.Pp.version tag)
+          else
+            Prompt.confirm_or_abort_neg ~yes ~question:(fun l ->
+                l
+                  "%a Tag %a does not point to the commit you've provided \
+                   (default: HEAD). Do you want to delete it anyways?"
+                  Fmt.(styled `Red string)
+                  "Warning:" Text.Pp.version tag)
+        in
+        question >>= fun () ->
+        Vcs.delete_tag ~dry_run repo tag >>| fun () ->
+        App_log.success (fun m -> m "Deleted tag %a" Text.Pp.version tag)
+      else
+        Ok
+          (App_log.success (fun apply_log_l ->
```

Comment:
I'm wondering whether this should be a success or status since nothing actually happened. What do you think?

Reply:
Yes, status definitely seems more appropriate here. I've also replaced `success` by `status` in the case of trying to create a tag that already exists providing a "right" commit-ish.